### PR TITLE
Changes to go-client for ability to add enhanced_lag_policy attribute using resource[aci_epg_to_domain]

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: models/fv_ae_pg_lag_pol_att.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package models
```
Comment (lhercot) [inline models/fv_ae_pg_lag_pol_att.go:1]: name of file should be fv_aepg_lag_pol_att.go

Comment (lhercot) [reply models/fv_ae_pg_lag_pol_att.go:1]: I thought we had fixed that. Can you open an issue against autogen for this issue?

File: models/fv_ae_pg_lag_pol_att.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	"strconv"
+
+	"github.com/ciscoecosystem/aci-go-client/container"
+)
+
+const (
+	DnfvAEPgLagPolAtt        = "uni/tn-%s/ap-%s/epg-%s/rsdomAtt-[%s]/epglagpolatt"
+	RnfvAEPgLagPolAtt        = "epglagpolatt"
+	ParentDnfvAEPgLagPolAtt  = "uni/tn-%s/ap-%s/epg-%s/rsdomAtt-[%s]"
+	FvaepglagpolattClassName = "fvAEPgLagPolAtt"
```
Comment (lhercot) [inline models/fv_ae_pg_lag_pol_att.go:14]: FvAEPgLagPolAttClassName

File: models/fv_ae_pg_lag_pol_att.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	Annotation string `json:",omitempty"`
+}
+
+func NewApplicationEPGLagPolicy(fvAEPgLagPolAttRn, parentDn string) *ApplicationEPGLagPolicy {
+	dn := fmt.Sprintf("%s/%s", parentDn, fvAEPgLagPolAttRn)
+	return &ApplicationEPGLagPolicy{
+		BaseAttributes: BaseAttributes{
+			DistinguishedName: dn,
+			Status:            "created, modified",
+			ClassName:         FvaepglagpolattClassName,
```
Comment (lhercot) [inline models/fv_ae_pg_lag_pol_att.go:31]: FvAEPgLagPolAttClassName

File: models/fv_ae_pg_lag_pol_att.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	return fvAEPgLagPolAttMap, err
+}
+
+func ApplicationEPGLagPolicyFromContainerList(cont *container.Container, index int) *ApplicationEPGLagPolicy {
+	ApplicationEPGLagPolicyCont := cont.S("imdata").Index(index).S(FvaepglagpolattClassName, "attributes")
+	return &ApplicationEPGLagPolicy{
+		BaseAttributes{
+			DistinguishedName: G(ApplicationEPGLagPolicyCont, "dn"),
+			Status:            G(ApplicationEPGLagPolicyCont, "status"),
+			ClassName:         FvaepglagpolattClassName,
```
Comment (lhercot) [inline models/fv_ae_pg_lag_pol_att.go:51]: FvAEPgLagPolAttClassName

**Review by lhercot — CHANGES_REQUESTED** [review]
If you are adding the model file. Also add the Service in case anyone need to manipulate this object directly.

File: client/fvAEPgLagPolAtt_service.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/container"
+	"github.com/ciscoecosystem/aci-go-client/models"
+)
+
+func (sm *ServiceManager) CreateApplicationEPGLagPolicy(domain_tDn string, application_epg string, application_profile string, tenant string, description string, nameAlias string, fvAEPgLagPolAttAttr models.ApplicationEPGLagPolicyAttributes) (*models.ApplicationEPGLagPolicy, error) {
```
Comment (lhercot) [inline client/fvAEPgLagPolAtt_service.go:10]: domain_tDn -> domain_dn

File: client/fvAEPgLagPolAtt_service.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+	rn := fmt.Sprintf(models.RnfvAEPgLagPolAtt)
+	parentDn := fmt.Sprintf(models.ParentDnfvAEPgLagPolAtt, tenant, application_profile, application_epg, domain_tDn)
+	fvAEPgLagPolAtt := models.NewApplicationEPGLagPolicy(rn, parentDn, description, nameAlias, fvAEPgLagPolAttAttr)
+	err := sm.Save(fvAEPgLagPolAtt)
+	return fvAEPgLagPolAtt, err
+}
+
+func (sm *ServiceManager) ReadApplicationEPGLagPolicy(domain_tDn string, application_epg string, application_profile string, tenant string) (*models.ApplicationEPGLagPolicy, error) {
```
Comment (lhercot) [inline client/fvAEPgLagPolAtt_service.go:18]: domain_tDn -> domain_dn

File: client/fvAEPgLagPolAtt_service.go [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+	cont, err := sm.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+
+	fvAEPgLagPolAtt := models.ApplicationEPGLagPolicyFromContainer(cont)
+	return fvAEPgLagPolAtt, nil
+}
+
+func (sm *ServiceManager) DeleteApplicationEPGLagPolicy(domain_tDn string, application_epg string, application_profile string, tenant string) error {
```
Comment (lhercot) [inline client/fvAEPgLagPolAtt_service.go:30]: domain_tDn -> domain_dn

File: client/fvAEPgLagPolAtt_service.go [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+	dn := fmt.Sprintf(models.DnfvAEPgLagPolAtt, tenant, application_profile, application_epg, domain_tDn)
+	return sm.DeleteByDn(dn, models.FvaepglagpolattClassName)
+}
+
+func (sm *ServiceManager) UpdateApplicationEPGLagPolicy(domain_tDn string, application_epg string, application_profile string, tenant string, description string, nameAlias string, fvAEPgLagPolAttAttr models.ApplicationEPGLagPolicyAttributes) (*models.ApplicationEPGLagPolicy, error) {
```
Comment (lhercot) [inline client/fvAEPgLagPolAtt_service.go:35]: domain_tDn -> domain_dn

File: client/fvAEPgLagPolAtt_service.go [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+	rn := fmt.Sprintf(models.RnfvAEPgLagPolAtt)
+	parentDn := fmt.Sprintf(models.ParentDnfvAEPgLagPolAtt, tenant, application_profile, application_epg, domain_tDn)
+	fvAEPgLagPolAtt := models.NewApplicationEPGLagPolicy(rn, parentDn, description, nameAlias, fvAEPgLagPolAttAttr)
+	fvAEPgLagPolAtt.Status = "modified"
+	err := sm.Save(fvAEPgLagPolAtt)
+	return fvAEPgLagPolAtt, err
+}
+
+func (sm *ServiceManager) ListApplicationEPGLagPolicy(domain_tDn string, application_epg string, application_profile string, tenant string) ([]*models.ApplicationEPGLagPolicy, error) {
```
Comment (lhercot) [inline client/fvAEPgLagPolAtt_service.go:44]: domain_tDn -> domain_dn

File: models/fv_aepg_lag_pol_att.go [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+const (
+	DnfvAEPgLagPolAtt        = "uni/tn-%s/ap-%s/epg-%s/rsdomAtt-[%s]/epglagpolatt"
+	RnfvAEPgLagPolAtt        = "epglagpolatt"
+	ParentDnfvAEPgLagPolAtt  = "uni/tn-%s/ap-%s/epg-%s/rsdomAtt-[%s]"
+	FvAEPgLagPolAttClassName = "fvAEPgLagPolAtt"
+)
+
+type ApplicationEPGLagPolicy struct {
+	BaseAttributes
+	NameAliasAttribute
```
Comment (lhercot) [inline models/fv_aepg_lag_pol_att.go:19]: It does not seems that this class has NameAlias. Please check and cleanup if needed. (It also does not have description)

**Review by lhercot — CHANGES_REQUESTED** [review]

File: models/fv_aepg_lag_pol_att.go (hunk 10 above)
Comment (shrsr) [reply models/fv_aepg_lag_pol_att.go:19]: ok

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**